tests/post/steps/test_monitoring.py

```diff
@@ -81,6 +96,35 @@ def _check_object_exists():
     utils.retry(_check_object_exists, times=20, wait=3)
 
 
+# FIXME: make these fixtures / helpers generic and easier to share
+@given("I have created a test Volume")
+def test_volume(k8s_apiclient, ssh_config):
+    client = kube_utils.VolumeClient(
+        CustomObjectsApi(api_client=k8s_apiclient), ssh_config
+    )
+    random_salt = ''.join(
+        random.choice(string.ascii_lowercase) for _ in range(8)
+    )
+    volume_name = "test-volume-{}".format(random_salt)
+    client.create_from_yaml(DEFAULT_VOLUME.format(name=volume_name))
+
+    def _check_volume_ready():
+        volume = client.get(volume_name)
+        assert volume is not None, 'Volume not found'
+        assert 'status' in volume, 'Volume has no status'
+        phase = kube_utils.VolumeClient.compute_phase(volume['status'])
+        assert phase == 'Available', 'Volume not ready'
+        assert volume['status'].get('deviceName') is not None, \
+            'Volume status.deviceName has not been reconciled'
+        return volume
+
+    yield utils.retry(
+        _check_volume_ready, times=30, wait=2,
+        name='waiting for Volume {} to be ready'.format(volume_name)
+    )
+
+    client.delete(volume_name, sync=True)
```

Comment:
Can't we reuse the existing `@given(parsers.parse("a Volume '{name}' exist"))`
Looks very similar.

If not the step directly, at least the implementation (to reduce code duplication)

Reply:
Can do!

Reply:
Did something, but couldn't factorize everything (the test Volume doesn't have a static name, and the teardown is handled directly in the `given` fixture).

Reply:
Do we really need the random name?
We could use a simple naming scheme, like `test-monitoring-volumeN`?

Reply:
Most of the logic was factorized in the `VolumeClient` directly :slightly_smiling_face: but still not sharing the fixture/step because of how Volume tests rely on the `context` fixture.